File: truss-compiler/src/main/java/truss/compiler/support/ArgumentMultiple.java
package truss.compiler.support;

import java.util.ArrayList;
import java.util.List;

public class ArgumentMultiple extends Argument {
    private final List<String> values = new ArrayList<>();

    public ArgumentMultiple(String option, String description, boolean mandatory) {
        super(option, description, mandatory);
    }

    @Override
    public boolean allowMultiple() {
        return true;
    }

    @Override
    public boolean hasParameter() {
        return true;
    }

    @Override
    public void setValue(String value) {
        values.add(value);
    }

    public List<String> getValues() {
        return values;
    }
}

File: Truss.Compiler/Support/ArgumentMultiple.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Truss.Compiler.Support {
    public class ArgumentMultiple : Argument {
        public ArgumentMultiple(string option, string description, bool mandatory)
            : base(option, description, mandatory) {
            Values = new List<string>();
        }

        public override bool AllowMultiple {
            get { return true; }
        }

        public override bool HasParameter {
            get { return true; }
        }

        public override void SetValue(string value) {
            Values.Add(value);
        }

        public List<string> Values { get; private set; }
    }
}
